common/txmgr/types/tx.go

```diff
@@ -32,3 +37,75 @@ const (
 type Transaction interface {
 	GetID() string
 }
+
+type NewTx[ADDR types.Hashable, TX_HASH types.Hashable] struct {
+	FromAddress      ADDR
+	ToAddress        ADDR
+	EncodedPayload   []byte
+	GasLimit         uint32
+	Meta             *TxMeta[ADDR, TX_HASH]
+	ForwarderAddress ADDR
+
+	// Pipeline variables - if you aren't calling this from ethtx task within
+	// the pipeline, you don't need these variables
+	MinConfirmations  clnull.Uint32
+	PipelineTaskRunID *uuid.UUID
+
+	Strategy TxStrategy
+
+	// Checker defines the check that should be run before a transaction is submitted on chain.
+	Checker TransmitCheckerSpec[ADDR]
+}
+
+// TransmitCheckerSpec defines the check that should be performed before a transaction is submitted
+// on chain.
+type TransmitCheckerSpec[ADDR types.Hashable] struct {
+	// CheckerType is the type of check that should be performed. Empty indicates no check.
+	CheckerType TransmitCheckerType `json:",omitempty"`
+
+	// VRFCoordinatorAddress is the address of the VRF coordinator that should be used to perform
+	// VRF transmit checks. This should be set iff CheckerType is TransmitCheckerTypeVRFV2.
+	VRFCoordinatorAddress *ADDR `json:",omitempty"`
+
+	// VRFRequestBlockNumber is the block number in which the provided VRF request has been made.
+	// This should be set iff CheckerType is TransmitCheckerTypeVRFV2.
+	VRFRequestBlockNumber *big.Int `json:",omitempty"`
+}
+
+// TransmitCheckerType describes the type of check that should be performed before a transaction is
+// executed on-chain.
+type TransmitCheckerType string
+
+// TxMeta contains fields of the transaction metadata
+// Not all fields are guaranteed to be present
+type TxMeta[ADDR types.Hashable, TX_HASH types.Hashable] struct {
```

Comment:
This approach for this struct sounds good.
However just pointing out 1 thing. This struct is not used for any logical purposes.
This struct is dumped as it is to DB, and then, when loaded in-memory, it is logged as it is.
The services teams use this struct only for logging purposes, to track their requests inside Txmgr.

So given all of that, we could convert everything here into just strings too. These strings can be logged as it is.
It also makes cleaner for reader, that nothing here is chain-specific, and we don't need to carefully generalize this struct.

Reply:
Without digging too deep into how `TxMeta` is used my worry is that it removes some useful functionality for types - where certain type methods might be used.

At a quick glance it seems like it's used in the job pipelines too - so not just logging. If it were all strings, it might require inserting unmarshalling lines to convert from strings to types rather than having them in one place and having expected types returned

Reply:
Yes, not asking to change any type for now.
Jobs pipeline may be just setting these fields, at the time of request creation.
The txmgr never returns this struct to anyone, so can't be used anywhere outside.